### ISSUE TITLE
Addresses fail to become ready if globalMaxSize is omitted from the infraconfig

### ISSUE THREAD

**Issue by k-wall** [EnMasseProject/enmasse#5130]
**Describe the bug**

If globalMaxSize is omitted from the standardinfraconfig or brokeredinfraconfig, addresses associated with addressspaces using that infraconfig will fail to become ready.

Internally, the agent fails like this:

```
2020-08-11T15:19:22.832Z agent INFO  [broker-4fdeff9-seeu-0] all connectors exist
2020-08-11T15:19:22.841Z agent INFO  [broker-4fdeff9-seeu-0] Adding/updating address settings: myqueue
2020-08-11T15:19:22.847Z agent ERROR [broker-4fdeff9-seeu-0] Failed to create new address setting myqueue: null
2020-08-11T15:19:22.847Z agent ERROR [broker-4fdeff9-seeu-0] Failed to retrieve broker address setting: myqueue null
2020-08-11T15:19:22.847Z agent ERROR [broker-4fdeff9-seeu-0] failed to retrieve addresses: null
2020-08-11T15:19:22.847Z agent WARN  Rescheduling failed serialized func (null) in 5000ms.
```

Tracing the Agent's interaction with the Broker, the addAddressSetting management function fails on the Broker with  `java.lang.IllegalArgumentException`.    Examining with a debugger reveals that the maxSizeByte argument is null.  The reason this is null is because the Agent's calculation has produced a `NaN` value.   The `NaN` value occurs when the `globalMaxSize` is not passed via the  `BROKER_GLOBAL_MAX_SIZE` environment variable.

```
java.lang.IllegalArgumentException
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.apache.activemq.artemis.core.server.management.impl.ManagementServiceImpl.invokeOperation(ManagementServiceImpl.java:817)
        at org.apache.activemq.artemis.core.server.management.impl.ManagementServiceImpl.handleMessage(ManagementServiceImpl.java:441)
        at org.apache.activemq.artemis.core.server.impl.ServerSessionImpl.handleManagementMessage(ServerSessionImpl.java:2034)
        at org.apache.activemq.artemis.core.server.impl.ServerSessionImpl.send(ServerSessionImpl.java:1803)
```

**To Reproduce**

**Expected behavior**

**Screenshots**

**Additional context**

**Comment by vbusch** [comment]
The simplest solution is to make global max size a required field.  This would involve:
- making it mandatory
- clean up unused code 'Artemis.prototype.getGlobalMaxSize' function and callers
- update docs

@k-wall @lulf are there any issues in making it required?

If so, I will fix the existing code.  When it is omitted, if maxSizeBytes isn't set, it can get the global max size from the broker, and use it for the calculation.

**Comment by lulf** [comment]
@vbusch i think that would break existing infra configs for a lot of users if we made it mandatory, so we cannot do that. How about deducing it based on broker memory size of the infra config. It should be 1/8 of broker memory by default.

**Comment by vbusch** [comment]
@lulf making it mandatory is out then.  
Doing 1/8th of the broker memory size of the infra config, would save us a call to the broker to request its globalMaxSize.  But I don't think that one call is costing us much.
I think requesting the globalMaxSize from the broker, would do a better job of getting a more accurate number.  I believe the broker defaults global max size to 1/2 the jvms memory  (but will have to test it manually, since I am not seeing it in the docs).

**Comment by lulf** [comment]
@vbusch I'm fine either way. However, we have modified the broker startup scripts to detect '-1' and convert that to 1/8th of the container memory rather than using the broker internal defaults of 1/4 of container memory , which had problems in some edge cases.

So default should be 1/8th no matter what if you get it from the broker or deduce it from looking at the config.

**Comment by vbusch** [comment]
In the broker startup scripts, if globalMaxSize wasn't set, it was defaulting it to 1/8th of the container memory of the init container.  The init container memory was defaulting to petabytes even if the broker container was going to be much smaller (as set in the infra config).
I have modified the startup scripts to get the broker container memory instead of the init container memory.  This helps when the infra config has the memory set for the broker.

The infraConfig broker memory isn't a required field.  In the case where it isn't set, in crc, the broker container memory is 9923 petabytes.  Which is giving an unrealistically huge globalMaxSize.

I have switched back to getting the globalMaxSize from the broker, instead of the infraconfig broker memory, since the latter isn't mandatory, and might not be set.
